### DISCUSSION TITLE
[BUG] Channel map variable not assigning to script variable in post processor script.

### DISCUSSION THREAD

**Discussion by jamesNWT** [nextgenhealthcare/connect#6242] [Q&A] [answered]
**Describe the bug**
I set a variable in my post processor script to a channel map variable via `var errorMessage = $('errorMessage');`, however, when I the immediately call `logger.info(errorMessage)` I get blank output, while `logger.info($('errorMessage'))` gives me my expected string.

**To Reproduce**
I can't reproduce this with other variable in the same channel. I am also too much of a mirth beginner to quickly set up a new test channel to see if I can do this again with a minimal channel. However, I will give a brief description of how the channel sets and gets the variable.
1.The variable is set in the response script of an http sender destination. The input/output of the destination is set to raw so the script runs every time. 
2. The variable is retrieved in the postProcessing script of the channel as described above.

**Expected behavior**
Setting a variable in the script to the channel map variable should allow me to reference the script variable and get the same value as the channel map variable.

**Actual behavior**
Script variable is empty string while channel map variable is expected string.

**Screenshots**
![image](https://github.com/nextgenhealthcare/connect/assets/50497892/46be442c-fea8-4a2a-8fb1-6557c8be99ec)
![image](https://github.com/nextgenhealthcare/connect/assets/50497892/8bfc3de4-6740-4bf8-a248-0b8067f6bad9)

**Environment (please complete the following information):**
 - OS: Windows Server 2016
 - Java Distribution/Version: openjdk 17.0.7 2023-04-18 LTS
 - Connect Version 4.4.1

**Workaround(s)**
I can for now just reference the channel map variable directly without storing in the script variable.

**Additional context**
Apologies if this is a bad report, I've been thrown into mirth without much training and don't really know what I don't know.

**Comment by jonbartels** [comment, marked as answer]
I have highlighted your problem in orange:
![image](https://github.com/nextgenhealthcare/connect/assets/7669292/4684a800-178c-49c0-a643-4f152f77f18d)

You set `errorMessage` from the map value then immediately reset it to empty string.

> Apologies if this is a bad report, I've been thrown into mirth without much training and don't really know what I don't know.

I've been using the software since 2007 I think, and I feel that every day!

**Reply by jamesNWT** [reply]
Wow I'm humiliated. Thanks.

**Comment by jamesNWT** [comment]
holy moly. this is what 7 straight hours looking at JavaScript does to your brain.

**Reply by pacmano1** [reply]
Well, don't beat up yourself, this is probably what you are thinking about.

```JS
var somevalue = {
	"stuff" : "that"
}
logger.info("somevalue: " + somevalue.stuff)
$c('somevalue',somevalue)
somevalue.stuff = "newstuff"
logger.info($c('somevalue').stuff)
```
![image](https://github.com/nextgenhealthcare/connect/assets/44065187/74b126b8-51f9-4709-afa1-07a5f6e3e6c0)
